service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -276,4 +286,67 @@ private void buildCoreItem(String instanceId, String barcode, NodeList nodes, Ma
         catalogItems.put(barcode, itemData);
     }
 
+    @Override
+    public FeesFines getFeesFines(String uin) {
+        String path = "patron/account/";
+        String additional = "&includeLoans=false&includeCharges=true&includeHolds=false";
+        String url = String.format("%s%s%s?apikey={apikey}%s", getAPIBase(), path, uin, additional);
+        String apiKey = getAuthentication().get(CatalogServiceFactory.FIELD_APIKEY);
+
+        logger.debug(String.format("Asking for fines from: %s", url));
+
+        JsonNode node = restTemplate.getForObject(url, JsonNode.class, apiKey);
+
+        Double total = null;
+        if (node.has("totalCharges") && node.get("totalCharges").has("amount")) {
+            total = node.get("totalCharges").get("amount").asDouble();
+        }
+
+        List<FeeFine> list = new ArrayList<>();
+
+        if (node.has("charges")) {
+            JsonNode charges = node.get("charges");
+
+            charges.forEach((JsonNode charge) -> {
+                double amount = 0;
+                if (charge.has("chargeAmount") && charge.get("chargeAmount").has("amount")) {
+                    amount = charge.get("chargeAmount").get("amount").asDouble();
+                }
+
+                String fineId = charge.has("feeFineId") ? charge.get("feeFineId").asText() : null;
+                String type = charge.has("reason") ? charge.get("reason").asText() : null;
+                Date date = charge.has("accrualDate") ? folioDateToDate(charge.get("accrualDate").asText()) : null;
+
+                String title = null;
+                if (charge.has("item") && charge.get("item").has("title")) {
+                    title = charge.get("item").get("title").asText();
+                }
+
+                list.add(new FeeFine(amount, fineId, type, date, title));
+            });
+        }
+
+        return new FeesFines(uin, total, list.size(), list);
+    }
+
+    /**
+     * Convert from Folio dates, "yyyy-MM-dd'T'HH:mm:ss.SSSZ", to the Java Date.
+     *
+     * @param folioDate
+     * @return
+     * @throws ParseException
+     */
+    private Date folioDateToDate(String folioDate) {
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+        formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        try {
+            return Date.from(formatter.parse(folioDate).toInstant());
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        return null;
```

Comment:
I am not a big fan of returning null. Would it make more sense to throw the exception and handle it elsewhere? This also looks like a very general method and likely should be a static utility method. Maybe in a DateUtility. It could either standardize date formatting or afford an additional argument to provide the format.